File: gherkin/dotnet/Gherkin/Ast/DocString.cs
namespace Gherkin.Ast
{
    public class DocString : StepArgument, IHasLocation
    {
        public Location Location { get; private set; }
        public string ContentType { get; private set; }
        public string Content { get; private set; }
        public string Delimiter { get; private set; }

        public DocString(Location location, string contentType, string content, string delimiter = null)
        {
            Location = location;
            ContentType = contentType;
            Content = content;
            Delimiter = delimiter;
        }
    }
}

File: gherkin/java/src/main/java/gherkin/ast/DocString.java
package gherkin.ast;

public class DocString extends Node {
    private final String contentType;
    private final String content;

    public DocString(Location location, String contentType, String content) {
        super(location);
        this.contentType = contentType;
        this.content = content;
    }

    public String getContent() {
        return content;
    }

    public String getContentType() {
        return contentType;
    }
}
